TestDevPortForward is flaky

https://travis-ci.com/GoogleContainerTools/skaffold/jobs/230109409

```
--- FAIL: TestDevPortForward (146.37s)
    dev_test.go:201: waiting for connection...
    dev_test.go:296: event received timestamp:<seconds:1567436510 nanos:667158576 > event:<metaEvent:<entry:"Starting Skaffold: &{Version:v0.37.0-86-g4e37814 ConfigVersion:skaffold/v1beta15 GitVersion: GitCommit:4e37814bd0a4c5cf555b1f234a0a47d8476cc12d GitTreeState:clean BuildDate:2019-09-02T14:53:18Z GoVersion:go1.12.9 Compiler:gc Platform:linux/amd64}" > > 
    dev_test.go:296: event received timestamp:<seconds:1567436510 nanos:725595941 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-app" status:"In Progress" > > entry:"Build started for artifact gcr.io/k8s-skaffold/leeroy-app" 
    dev_test.go:296: event received timestamp:<seconds:1567436510 nanos:725924951 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-web" status:"In Progress" > > entry:"Build started for artifact gcr.io/k8s-skaffold/leeroy-web" 
    dev_test.go:296: event received timestamp:<seconds:1567436510 nanos:728803392 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-web" status:"Complete" > > entry:"Build completed for artifact gcr.io/k8s-skaffold/leeroy-web" 
    dev_test.go:296: event received timestamp:<seconds:1567436510 nanos:738410480 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-app" status:"Complete" > > entry:"Build completed for artifact gcr.io/k8s-skaffold/leeroy-app" 
    dev_test.go:296: event received timestamp:<seconds:1567436514 nanos:777816445 > event:<deployEvent:<status:"In Progress" > > entry:"Deploy started" 
    dev_test.go:296: event received timestamp:<seconds:1567436515 nanos:159111034 > event:<deployEvent:<status:"Complete" > > entry:"Deploy complete" 
    dev_test.go:287: event received timestamp:<seconds:1567436515 nanos:243112934 > event:<portEvent:<localPort:9000 remotePort:8080 namespace:"default" resourceType:"deployment" resourceName:"leeroy-web" > > entry:"Forwarding container  to local port 9000" 
    dev_test.go:287: event received timestamp:<seconds:1567436515 nanos:246233817 > event:<portEvent:<localPort:50051 remotePort:50051 namespace:"skaffoldspths" resourceType:"service" resourceName:"leeroy-app" > > entry:"Forwarding container  to local port 50051" 
    dev_test.go:292: Detected service/leeroy-app is forwarded to port 50051
    dev_test.go:296: event received timestamp:<seconds:1567436519 nanos:317692442 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-app" status:"In Progress" > > entry:"Build started for artifact gcr.io/k8s-skaffold/leeroy-app" 
    dev_test.go:296: event received timestamp:<seconds:1567436519 nanos:319870088 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-app" status:"In Progress" > > entry:"Build started for artifact gcr.io/k8s-skaffold/leeroy-app" 
    dev_test.go:296: event received timestamp:<seconds:1567436522 nanos:372002592 > event:<buildEvent:<artifact:"gcr.io/k8s-skaffold/leeroy-app" status:"Complete" > > entry:"Build completed for artifact gcr.io/k8s-skaffold/leeroy-app" 
    dev_test.go:296: event received timestamp:<seconds:1567436524 nanos:377053678 > event:<deployEvent:<status:"In Progress" > > entry:"Deploy started" 
    dev_test.go:296: event received timestamp:<seconds:1567436524 nanos:647508646 > event:<deployEvent:<status:"Complete" > > entry:"Deploy complete" 
    dev_test.go:287: event received timestamp:<seconds:1567436524 nanos:655748164 > event:<portEvent:<localPort:50051 remotePort:50051 namespace:"skaffoldspths" resourceType:"service" resourceName:"leeroy-app" > > entry:"Forwarding container  to local port 50051" 
    dev_test.go:292: Detected service/leeroy-app is forwarded to port 50051
    dev_test.go:316: Timed out waiting for response from port 50051
```

--- a/integration/dev_test.go
+++ b/integration/dev_test.go
@@ -305,7 +305,8 @@ func waitForPortForwardEvent(t *testing.T, entries chan *proto.LogEntry, resourc
 
 // assertResponseFromPort waits for two minutes for the expected response at port.
 func assertResponseFromPort(t *testing.T, port int, expected string) {
-	t.Logf("Waiting for response %s from port %d", expected, port)
+	url := fmt.Sprintf("http://%s:%d", util.Loopback, port)
+	t.Logf("Waiting on %s to return: %s", url, expected)
 	ctx, cancelTimeout := context.WithTimeout(context.Background(), 2*time.Minute)
 	defer cancelTimeout()
 
@@ -314,21 +315,22 @@ func assertResponseFromPort(t *testing.T, port int, expected string) {
 		case <-ctx.Done():
 			t.Fatalf("Timed out waiting for response from port %d", port)
 		case <-time.After(1 * time.Second):
-			resp, err := http.Get(fmt.Sprintf("http://%s:%d", util.Loopback, port))
+			client := http.Client{Timeout: 1 * time.Second}
+			resp, err := client.Get(url)
 			if err != nil {
-				t.Logf("error getting response from port %d: %v", port, err)
+				t.Logf("[retriable error]: %v", err)
 				continue
 			}
 			defer resp.Body.Close()
 			body, err := ioutil.ReadAll(resp.Body)
 			if err != nil {
-				t.Logf("error reading response: %v", err)
+				t.Logf("[retriable error] reading response: %v", err)
 				continue
 			}
 			if string(body) == expected {
 				return
 			}
-			t.Logf("didn't get expected response from port. got: %s, expected: %s", string(body), expected)
+			t.Logf("[retriable error] didn't get expected response from port. got: %s, expected: %s", string(body), expected)
 		}
 	}
 }

--- a/integration/dev_test.go
+++ b/integration/dev_test.go
@@ -26,7 +26,6 @@ import (
 	"testing"
 	"time"
 
-	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/tools/clientcmd"
 
@@ -306,7 +305,7 @@ func waitForPortForwardEvent(t *testing.T, entries chan *proto.LogEntry, resourc
 
 // assertResponseFromPort waits for two minutes for the expected response at port.
 func assertResponseFromPort(t *testing.T, port int, expected string) {
-	logrus.Infof("Waiting for response %s from port %d", expected, port)
+	t.Logf("Waiting for response %s from port %d", expected, port)
 	ctx, cancelTimeout := context.WithTimeout(context.Background(), 2*time.Minute)
 	defer cancelTimeout()
 
@@ -317,19 +316,19 @@ func assertResponseFromPort(t *testing.T, port int, expected string) {
 		case <-time.After(1 * time.Second):
 			resp, err := http.Get(fmt.Sprintf("http://%s:%d", util.Loopback, port))
 			if err != nil {
-				logrus.Infof("error getting response from port %d: %v", port, err)
+				t.Logf("error getting response from port %d: %v", port, err)
 				continue
 			}
 			defer resp.Body.Close()
 			body, err := ioutil.ReadAll(resp.Body)
 			if err != nil {
-				logrus.Infof("error reading response: %v", err)
+				t.Logf("error reading response: %v", err)
 				continue
 			}
 			if string(body) == expected {
 				return
 			}
-			logrus.Infof("didn't get expected response from port. got: %s, expected: %s", string(body), expected)
+			t.Logf("didn't get expected response from port. got: %s, expected: %s", string(body), expected)
 		}
 	}
 }

